File: nodes/range/range.process.js
export default async ({inputs, settings, config}) => {

  const numbers = Array.isArray(inputs.numbers) ? inputs.numbers : [inputs.numbers];
  const validNumbers = numbers.map(n => Number(n)).filter(n => !isNaN(n));
  
  if (validNumbers.length === 0) {
    return {
      min: 0,
      max: 0,
      range: 0
    };
  }
  
  const min = Math.min(...validNumbers);
  const max = Math.max(...validNumbers);
  
  return {
    min,
    max,
    range: max - min
  };
}; 

File: nodes/range/range.process.py
async def process(inputs, settings, config, nodeConfig):
  numbers = inputs.get("numbers", [])
  if not isinstance(numbers, list):
    numbers = [numbers]
    
  valid_numbers = []
  for n in numbers:
    try:
      num = float(n)
      if not isinstance(num, complex) and not isinstance(n, bool):
        valid_numbers.append(num)
    except (ValueError, TypeError):
      continue
  
  if not valid_numbers:
    return {
      "min": 0,
      "max": 0,
      "range": 0
    }
    
  min_val = min(valid_numbers)
  max_val = max(valid_numbers)
  
  return {
    "min": min_val,
    "max": max_val,
    "range": max_val - min_val
  } 